packages/atcommons/src/enroll_namespace.c

```diff
@@ -5,19 +5,27 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-
+#include <stdint.h>
 #include <atlogger/atlogger.h>
 
 #define TAG "enroll_namespace"
 
 int atcommons_enroll_namespace_list_append(atcommons_enroll_namespace_list_t **ns_list,
                                            atcommons_enroll_namespace_t *ns) {
-  // allocate enough memory for enroll_namespace_list struct, and the number of atcommons_enroll_namespace_t structs
-  // that are in the list
-  atcommons_enroll_namespace_list_t *temp =
-      realloc(*ns_list, sizeof(atcommons_enroll_namespace_list_t) +
-                            sizeof(atcommons_enroll_namespace_t) * ((*ns_list)->length + 1));
+  if (ns == NULL) {
+    atlogger_log(TAG, 0, "Namespace to append cannot be null\n");
+    return -1;
+  }
+
+  // If the list's length is uninitialized (SIZE_MAX), set it to 0
+  if ((*ns_list)->length == SIZE_MAX) {
+    (*ns_list)->length = 0;
+  }
```

Comment:
If the list length is not 0, that means the struct was never 0'd out, which is our recommended practice to avoid unexpected initialization values from being used. This means an init function needs to be added, or memset on the struct needs to be called in that init function.

Reply:
This has been addressed in [acdc3ea](https://github.com/atsign-foundation/at_c/pull/453/commits/acdc3ea336f28a85dc16dc042d47ca1af2c61968)